### PR TITLE
mockhyperbahn: use tchannel relaying to implement in-memory forwarding

### PR DESCRIPTION
**Pull request by prashantv**
Now that we have inbuilt relaying and ability to handle calls on relays, we can implement a simple in-memory Hyperbahn in the mockhyperbahn package!

### CODE REVIEW

File: testutils/mockhyperbahn/advertise_table_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package mockhyperbahn
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/uber/tchannel-go/testutils"
```
Comment (abhinav) [inline testutils/mockhyperbahn/advertise_table_test.go]: import group ordering

**Comment by abhinav** [comment]
LGTM with nit

File: testutils/mockhyperbahn/advertise_table_test.go (hunk 1 above)
Comment (prashantv) [inline testutils/mockhyperbahn/advertise_table_test.go]: Fixed

**Comment by abhinav** [comment]
:shipit:

File: testutils/mockhyperbahn/advertise_table.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		return
+	}
+
+	hostPorts = append(hostPorts, hostPort)
+	sort.Strings(hostPorts)
+	t.serviceMapping[svc] = hostPorts
+}
+
+func (t *advertisedTable) Get(call relay.CallFrame) (relay.Peer, error) {
+	peers := t.serviceMapping[string(call.Service())]
```
Comment (akshayjshah) [inline testutils/mockhyperbahn/advertise_table.go]: Don't we need to hold the lock here?

**Comment by akshayjshah** [comment]
otherwise lgtm

File: testutils/mockhyperbahn/advertise_table.go (hunk 2 above)
Comment (prashantv) [inline testutils/mockhyperbahn/advertise_table.go]: Good catch! Fixed

**Pull request merged**